### PR TITLE
Easi 2611/plan basics to user table 

### PR DESCRIPTION
**Pull request by StevenWadeOddball**
# EASI-2611

## Changes and Description

- base task list section to use the user table
    - Generic Functionality to get the user account information for the readyForReviewByUserAccount and readyForClearancebyUserAccount.  
- Audit table updated to use the user table config

## How to test this change
1. Make a few model plans
2. Update a few plan basics, (optionally as different users so you can verify that multiple user Accounts are returned
3. Run this query to get all model plans with the UserAccounts represented by different UserIds
```gql
query modelPlanCollection {
    modelPlanCollection(filter: INCLUDE_ALL) {
        id
        basics{
            modelPlanID
            id      
            createdByUserAccount 
            {
                commonName
            }   
            modifiedByUserAccount
            {
                commonName
            }     
            readyForReviewByUserAccount 
            {
                commonName
            }
            readyForClearanceByUserAccount 
            {
                commonName
            }
        }
        modelName
        archived
        createdBy
        createdDts
        modifiedBy
        modifiedDts
        status
        isFavorite
        isCollaborator
        collaborators {
            createdByUserAccount 
            {
                commonName
                familyName
                givenName
                email
                locale
            }   
            userAccount 
            {
                commonName
                familyName
                givenName
                email
                locale
            }
            modifiedByUserAccount
            {
                commonName
                familyName
                givenName
                email
                locale
            }  
        }

        
        # plan
    }
}
```

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: pkg/models/base_task_list_section.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ type IBaseTaskListSection interface {
 	// GetBaseTaskListSection() *baseTaskListSection
 	CalcStatus(TaskStatus) error
 	//methods from BaseStruct
-	GetBaseStruct() *baseStruct
+	// GetBaseStruct() *baseStruct
```
Comment (OddTomBrooks) [inline pkg/models/base_task_list_section.go:16]: Is there a reason to keep this line?

File: pkg/models/base_task_list_section_user_table.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+// GetStatus returns the status of a basesTaskListSection
+func (b *baseTaskListSectionUserTable) GetStatus() TaskStatus {
+	return b.Status
+}
+
+// // GetBaseStruct returns the Base Struct
+// func (b *baseTaskListSectionUserTable) GetBaseStruct() *baseStruct {
+// 	return b.baseStructUserTable
+// }
```
Comment (OddTomBrooks) [inline pkg/models/base_task_list_section_user_table.go:49]: Do we need to keep this comment?

File: pkg/models/base_task_list_section_user_table.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+// CalcStatus updates the TaskStatus if it is in ready, but it has been modified.
+func (b *baseTaskListSectionUserTable) CalcStatus(oldStatus TaskStatus) error {
+
+	// If model is moved out of READY_FOR_CLEARANCE it should be set to IN_PROGRESS
+	if oldStatus == TaskReadyForClearance && b.Status != TaskReadyForClearance {
+		b.Status = TaskInProgress
+		return nil
+	}
```
Comment (OddTomBrooks) [inline pkg/models/base_task_list_section_user_table.go:58]: Is there a case where the model can be moved out of READY_FOR_CLEARANCE and skip the IN_PROGRESS status? If so, the model status will not be properly calculated until the next evaluation.

**Review by OddTomBrooks — CHANGES_REQUESTED** [review]

File: pkg/models/base_task_list_section_user_table.go (hunk 3 above)
Comment (StevenWadeOddball) [reply pkg/models/base_task_list_section_user_table.go:58]: @OddTomBrooks , good call out, I actually had this same conversation yesterday. This is the expected behavior for now (it's just the same logic that exists in the other non-user table model).

File: pkg/models/base_task_list_section.go (hunk 1 above)
Comment (StevenWadeOddball) [reply pkg/models/base_task_list_section.go:16]: good call out, I'll refactor that out!

File: pkg/models/base_task_list_section_user_table.go (hunk 2 above)
Comment (StevenWadeOddball) [reply pkg/models/base_task_list_section_user_table.go:49]: Good call! I think it should be removed.

**Pull request merged**